build.assets/robotest/nightly_config.sh

```diff
@@ -21,15 +21,21 @@ UPGRADE_MAP[6.2.0]="ubuntu:16"
 
 UPGRADE_VERSIONS=${!UPGRADE_MAP[@]}
 
+# The following breaks a dependency loop. We need upgrade versions to generate tarballs,
+# but we need tarball names to generate the full test config.
+if [[ ${1} == "upgradeexit" ]]  ; then
```

Comment:
I'm not sure what this is for. Could you explain in more detail?

Reply:
I added extra documentation on these config files & how they're used in ccff414d9431d0e51400df43114baa851c31d8ee.  Check out `assets/robotest/config/README.md` and see if it sufficiently clarifies the implementation.